File: Python/polLegendre.py
import sympy as sp


def polLegendre(n):
    x = sp.Symbol("x")
    p0, p1 = 1, x

    for i in range(2, n+1):
        pk = sp.simplify(((2 * i - 1) * x * p1 - (i - 1) * p0) / i)
        p0 = p1
        p1 = pk

    return p1  


print(polLegendre(2))

File: Octave/polLegendre.m
function pk=polLegendre(n)
  pkg load symbolic
  syms x
  p0=1; p1=x;
  for k=2:n
    pk=simplify(((2*k-1)*x*p1-(k-1)*p0)/k);
    p0=p1;
    p1=pk;
  end
end

